### PR TITLE
Brings back Multipart from the dead!

### PR DESCRIPTION
**Pull request by felipecsl**
All tests should be passing. Re-enabled all Multipart related tests. Adapted old `RequestBuilder` class to work with new `OkHttp` classes. Everything seems to be working fine.
One minor details is that I wasnt able to assert on the Request body `endsWith()`, so I replaced them with `contains()` since we now just have a string instead of a list of parts. Maybe we can inject the boundary into `MultipartBuilder` somehow. Let me know!

### CODE REVIEW

File: retrofit/src/main/java/retrofit/RequestBuilder.java [hunk 1]
```diff
@@ -240,87 +266,90 @@ void setArguments(Object[] args) {
...
+//              throw new IllegalArgumentException(
+//                  "Parameter #" + (i + 1) + " field map contained null key.");
+//            }
+//            Object entryValue = entry.getValue();
+//            if (entryValue != null) { // Skip null values.
+//              formBody.addField(entryKey.toString(), encodeNames, entryValue.toString(),
+//                  encodeValues);
+//            }
+//          }
+//        }
```
Comment (JakeWharton) [inline retrofit/src/main/java/retrofit/RequestBuilder.java]: Can you revert the comment changes here? It adds needless noise to the commit which messes with history and prevent cherry-picking and reverting (not that I expect those to need to happen).

File: retrofit/src/test/java/retrofit/RequestBuilderTest.java [hunk 2]
```diff
@@ -1218,372 +1224,370 @@ Response method(@Path("ping") String ping, @Body Object body, @Path("kit") Strin
...
+//        return null;
+//      }
+//    }
+//    try {
+//      buildRequest(Example.class);
+//      fail();
+//    } catch (IllegalArgumentException e) {
+//      assertThat(e).hasMessage("Example.method: @FieldMap parameter type must be Map. (parameter #1)");
+//    }
+//  }
```
Comment (JakeWharton) [inline retrofit/src/test/java/retrofit/RequestBuilderTest.java]: Same here.

**Comment by JakeWharton** [comment]
I think this looks good. I'm not at home for a detailed review, but nothing jumped out.

**Comment by felipecsl** [comment]
Done. 
Build seems to be failing for this reason: `/home/travis/build/square/retrofit/retrofit/src/main/java/retrofit/RequestBuilder.java:218:3: Method length is 151 lines (max allowed is 150).` Should I do anything about it?
Also, do you want me to squash the commits in a single one?

**Comment by JakeWharton** [comment]
@jacobtabak comment on the PR diff view not the commit. No one gets emails for commit comments so I didn't even realize you saw this.

@felipecsl ignore it. I'll disable when I merge. You can squash or I can when I merge.

File: retrofit/src/test/java/retrofit/RequestBuilderTest.java [hunk 3]
```diff
@@ -1218,192 +1224,190 @@ Response method(@Path("ping") String ping, @Body Object body, @Path("kit") Strin
...
+
+    assertThat(bodyString)
+        .contains("name=\"kit\"")
+        .contains("\r\nkat\r\n--");
+  }
+
+  @Test public void multipartWithEncoding() throws IOException {
+    class Example {
+      @Multipart //
+      @POST("/foo/bar/") //
```
Comment (jacobtabak) [inline retrofit/src/test/java/retrofit/RequestBuilderTest.java]: What's up with these empty comments at the end of the line?  I see they're carried over from a previous commit.

Comment (JakeWharton) [inline retrofit/src/test/java/retrofit/RequestBuilderTest.java]: They prevent dumb automatic formatting (collapsing together) when you reformat the whole file.

File: retrofit/src/main/java/retrofit/RequestBuilder.java [hunk 4]
```diff
@@ -282,45 +308,49 @@ void setArguments(Object[] args) {
...
-      //              converter.toBody(entryValue, entryValue.getClass()));
-      //        }
-      //      }
-      //    }
-      //  }
+      } else if (annotationType == Part.class) {
+        if (value != null) { // Skip null values.
+          String name = ((Part) annotation).value();
+          String transferEncoding = ((Part) annotation).encoding();
+          Headers headers = Headers.of(
```
Comment (JakeWharton) [inline retrofit/src/main/java/retrofit/RequestBuilder.java]: Nice

**Comment by JakeWharton** [comment]
Fixed, squashed, merged. Thanks!

**Comment by felipecsl** [comment]
:+1: :+1:

**Pull request closed without merging**